integration-tests/smoke/keeper_test.go

```diff
@@ -221,6 +227,92 @@ var _ = Describe("Keeper Suite @keeper", func() {
 		err = chainClient.WaitForEvents()
 		Expect(err).ShouldNot(HaveOccurred(), "Error creating keeper jobs")
 
+		if testToRun == UpdateCheckDataTest {
+			By("tests that check data will be updated")
+
+			var newCheckData = []byte("abcde")
+			for i := 0; i < len(upkeepIDs); i++ {
+				err = registry.UpdateCheckData(upkeepIDs[i], newCheckData)
+				Expect(err).ShouldNot(HaveOccurred(), "Could not set check data for upkeep at index "+strconv.Itoa(i))
+			}
+
+			err = chainClient.WaitForEvents()
+			Expect(err).ShouldNot(HaveOccurred(), "Error encountered when waiting for check data update")
+
+			// retrieve old check data for all upkeeps
+			for i := 0; i < len(upkeepIDs); i++ {
+				upkeep, err := registry.GetUpkeepInfo(context.Background(), upkeepIDs[i])
+				Expect(err).ShouldNot(HaveOccurred(), "Failed to get upkeep info at index "+strconv.Itoa(i))
+				Expect(upkeep.CheckData).Should(Equal(newCheckData), "Expect the check data to be %s, but got %s", string(newCheckData), string(upkeep.CheckData))
+			}
```

Comment:
This is somewhat tested in contract unit tests. What would be good to test here is that the nodes start using the new checkData to perform upkeeps. 

E.g. you can have an upkeep which returns true only on specific checkData. You can test that the upkeep is not performed initially, but when the checkData changes it starts performing

Reply:
Done